src/PythonMigrationViewExtension/MigrationAssistant/PythonMigrationAssistantViewModel.cs

```diff
@@ -42,37 +43,49 @@ public IDiffViewViewModel CurrentViewModel
 
         public PythonMigrationAssistantViewModel(PythonNode pythonNode, WorkspaceModel workspace, IPathManager pathManager, Version dynamoVersion)
         {
-            this.PythonNode = pythonNode;
-            this.OldCode = pythonNode.Script;
+            PythonNode = pythonNode;
+            OldCode = pythonNode.Script;
 
             this.workspace = workspace;
-            this.backupDirectory = pathManager.BackupDirectory;
+            backupDirectory = pathManager.BackupDirectory;
             this.dynamoVersion = dynamoVersion;
 
-            MigrateCode();
+            try
+            {
+                MigrateCode();
+            }
+            catch (PythonException)
+            {
+                var sidebyside = new SideBySideDiffBuilder();
+                diffModel = sidebyside.BuildDiffModel(OldCode, OldCode, false);
+
+                SetSideBySideViewModel();
+                
+                CurrentViewModel.DiffState = State.Error;
+                return;
+            }
             SetSideBySideViewModel();
+
         }
 
         #region Code migration
 
         private void MigrateCode()
         {
-            this.NewCode = ScriptMigrator.MigrateCode(this.OldCode);
+            NewCode = ScriptMigrator.MigrateCode(OldCode);
 
             var sidebyside = new SideBySideDiffBuilder();
-            this.diffModel = sidebyside.BuildDiffModel(this.OldCode, this.NewCode, false);
+            diffModel = sidebyside.BuildDiffModel(OldCode, NewCode, false);
         }
 
         /// <summary>
         /// Replaces the code in the Python node with the code changes made by the Migration Assistant.
         /// </summary>
         public void ChangeCode()
         {
-            if (!this.CurrentViewModel.HasChanges)
+            if (CurrentViewModel.DiffState == State.Error || CurrentViewModel.DiffState == State.NoChanges)
```

Comment:
This code should not execute when the state is `Error` but please remove `CurrentViewModel.DiffState == State.Error` from here as it looks confusing. When the state is error, what's inside the `if` is something we don't want to do.